### PR TITLE
Init the CRDs and the controller

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Commands used:

operator-sdk init --domain=storage.openshift.io --verbose --repo github.com/kube-storage/volume-replication-operator
operator-sdk create api --group replication --kind=VolumeReplica --version=v1alpha1
operator-sdk create api --group replication --kind=VolumeReplicationClass --version=v1alpha1
make generate
make manifests

Signed-off-by: Raghavendra Talur <raghavendra.talur@gmail.com>

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
@sp98 @umangachapagain @iamniting @Madhu-1 Please take a look.

**Comment by Madhu-1** [comment]
> Commands used:
> 
> operator-sdk init --domain=storage.openshift.io --verbose --repo github.com/kube-storage/volume-replication-operator
> operator-sdk create api --group replication --kind=VolumeReplica --version=v1alpha1
> operator-sdk create api --group replication --kind=VolumeReplicationClass --version=v1alpha1
> make generate
> make manifests
> 
> Signed-off-by: Raghavendra Talur [raghavendra.talur@gmail.com](mailto:raghavendra.talur@gmail.com)

@raghavendra-talur can we make the kind as VolumeReplication which is a prefix of VolumeReplicationClass also?

* VolumeReplicationClass
* VolumeReplication

**Comment by sp98** [comment]
LGTM other than what Madhu has suggested about changing the Kind to `VolumeReplication`.

File: config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  creationTimestamp: null
+  name: volumereplicationclasses.replication.storage.openshift.io
+spec:
+  group: replication.storage.openshift.io
+  names:
+    kind: VolumeReplicationClass
+    listKind: VolumeReplicationClassList
+    plural: volumereplicationclasses
+    singular: volumereplicationclass
+  scope: Namespaced
```
Comment (umangachapagain) [inline config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml:17]: Shouldn't `VolumeReplicationClass` be Cluster Scoped?

Comment (Madhu-1) [reply config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml:17]: Yes it should be cluster scoped

File: api/v1alpha1/volumereplica_types.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// VolumeReplicaSpec defines the desired state of VolumeReplica
+type VolumeReplicaSpec struct {
```
Comment (iamniting) [inline api/v1alpha1/volumereplica_types.go:27]: I don't know if this code is autogenerated or not. If it is not then we should change these struct names also from `Replica` to `Replication`

**Review by iamniting — CHANGES_REQUESTED** [review]

File: api/v1alpha1/volumereplica_types.go (hunk 2 above)
Comment (sp98) [reply api/v1alpha1/volumereplica_types.go:27]: This is auto generated.

File: config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml (hunk 1 above)
Comment (ShyamsundarR) [reply config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml:17]: I think @raghavendra-talur is addressing as part of [this](https://github.com/kube-storage/volume-replication-operator/pull/3/commits/5b20f86ffbd35c8ec7da8cb1e4584ccb3bdccd2e#diff-7044490c8cb898854279e3ce0ff4f06a016d17e3513ab044d9eafd05398700c3R52) change.

**Comment by Madhu-1** [comment]
closing in favor of https://github.com/kube-storage/volume-replication-operator/pull/14

**Pull request closed without merging**